include/sampleflow/consumers/average_cosinus.h

```diff
@@ -41,81 +41,95 @@ namespace SampleFlow
-         *   simply ignores it.
-         */
-        virtual
-        void
-        consume (InputType     sample,
-                 AuxiliaryData aux_data) override;
-
-        /**
-         * A function that returns the average cosine vector computed from the
-         * samples seen so far. If no samples have been processed so far, then
-         * a default-constructed object of type std::vector<scalar_type> will be returned.
-         *
-         * @return The computed average cosine vector.
-         */
-        std::vector<scalar_type> get() const;
-
-      private:
-        /**
-         * A mutex used to lock access to all member variables when running
-         * on multiple threads.
-         */
-        mutable std::mutex mutex;
-
-        /**
-         * The data type, where we save multiple set of previous samples.
-         */
-        using matrix_type = boost::numeric::ublas::matrix<scalar_type>;
-
-        /**
-         * Current average cosine. Description of this is given above
-         */
-        std::vector<scalar_type> current_avg_cosine;
-
-        /**
-         * Save previous sample value
-         */
-        matrix_type previous_sample;
-        matrix_type previous_sample_replace;
-
-        /**
-         * The number of samples processed so far.
-         */
-        types::sample_index n_samples;
-
-        /**
-         * Describes how many values of average cosine function we calculate.
-         */
-        unsigned int history_length;
+    public:
+      /**
+       * The data type of the elements of the input type. This type is used to define type of members, used in
+       * calculations.
+       */
+      using scalar_type = types::ScalarType<InputType>;
+
+      /**
+       * The data type, where we save multiple set of previous samples.
+       */
+      using value_type = std::vector<scalar_type>;
+
+      /**
+       * Constructor.
+       *
+       * @param[in] lag_length A number that indicates how many average_cosinus
+       *   values we want to calculate, i.e., how far back in the past we
+       *   want to check how correlated (how big cosinus value) each sample is.
+       */
+      AverageCosineBetweenSuccessiveSamples(const unsigned int length);
+
+      /**
+       * Process one sample by updating the previously computed average cosine
+       * function using this one sample.
+       *
+       * @param[in] sample The sample to process.
+       * @param[in] aux_data Auxiliary data about this sample. The current
+       *   class does not know what to do with any such data and consequently
+       *   simply ignores it.
+       */
+      virtual
+      void
+      consume (InputType     sample,
+	       AuxiliaryData aux_data) override;
+
+      /**
+       * A function that returns the average cosine vector computed from the
+       * samples seen so far. If no samples have been processed so far, then
+       * a default-constructed object of value_type will be returned.
```

Comment:
Write this as follows to make sure we get nice markup when doxygen converts the documentation to HTML:
```suggestion
       * a default-constructed object of `value_type` will be returned.
```